src/lib/optimizer/abstract_syntax_tree/empty_node.hpp

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <string>
+
+#include "optimizer/abstract_syntax_tree/abstract_ast_node.hpp"
+
+namespace opossum {
+
+/**
+ * This node is returned by the SQLToASTTranslator for everything that doesn't have a return value (i.e., CREATE)
```

Comment:
What are those supposed to do? How is the `ASTToOperatorTranslator` supposed to translate them? If we had a `CreateViewNode`, that node would have to hold the AST that goes into the view.

Reply:
@mrzzzrm What do you think?

Reply:
Wow, I just realized we add the views in `SQLToASTTranslator::translate_parse_result()` - this is a big blocker, imo. The translator should NOT change the DB state and I would never expect it to from reading its interface or seeing code that invokes `SQLToASTTranslator::translate_parse_result()`. There should be appropriate Nodes and Operators for this. Only when executing the Operator, the View/Table should be created/dropped/etc.

Reply:
You might argue, that a `CreateViewOperator` that doesn't output a table is not in line with our Operator design. Not totally, but still better than this, imo.

Reply:
Like Moritz said, the views should not be added in the translator. So I think the correct way to do this would be to have nodes like `CreateViewNode` that are then translated into `CreateView` operators and then executed.